File: src/main/resources/automatureapi/automatureapi.rb
require 'socket'
require 'json'

class AutomatureAPI
host=""
sendport=0
receiveport = 0   

def initialize()
        @host =  '127.0.0.1'
        @sendport = Integer(ARGV[ARGV.length-1])
        @receiveport = Process.pid()    
    end
def alterContextVar(argv1,argv2)
  jsonarr = {"method" => "alter","name"=>argv1,"value"=>argv2,"port"=>"#{45000+Process.pid()}"}
  message=JSON.dump(jsonarr)
  
  sock = TCPSocket.new(@host, (45000+@sendport))
  sock.write(message)
  sock.close
  
  sock = TCPSocket.new(@host, 45000+@receiveport)
  s = sock.recv(1024)
  if JSON.parse(s)['error'].eql?"1"
   sock.close
   raise 'Error while altering context variable'  
  end
  
  sock.close
  
end

def getContextVar(argv1)
  jsonarr = {"method" => "get","name"=>argv1,"value"=>"","port"=>"#{45000+Process.pid()}"}
  message=JSON.dump(jsonarr)
  sock = TCPSocket.new(@host, (45000+@sendport))
  sock.write(message)
  sock.close
  
  sock = TCPSocket.new(@host, (45000+@receiveport))
  s = sock.recv(1024)

  if JSON.parse(s)['error'].eql?"1"
    sock.close
   raise 'Error while getting context variable'
  end
  s=JSON.parse(s)['value']
  sock.close
  return s
end
end

File: src/main/resources/automatureapi/automatureapi.py
import socket
import sys            
import json

class AutomatureAPI(object):  
  host = ""   
  port = 0  
  
  s=None
     
  def __init__(self):
  
    self.host="127.0.0.1"
   
    self.port=45000+int(sys.argv[len(sys.argv)-1]) % 10000
    
                 
  def alterContextVar(self,argv1,argv2): 
    self.s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    self.s.connect((self.host,self.port))
    message=("{\"method\":\"alter\",\"name\":\""+argv1+"\",\"value\":\""+argv2+"\"}")  
    self.s.sendall(bytes(message))
    data=self.s.recv(1024).decode("utf-8") 
    if json.loads( data)['error']=='1':                
         self.s.close()
         raise Exception('Error while altering context variable')  
    self.s.close()
    
  def getContextVar(self,argv1):
    self.s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    self.s.connect((self.host,self.port))
    message=("{\"method\":\"get\",\"name\":\""+argv1+"\",\"value\":\""+""+"\"}")  
    self.s.sendall(bytes(message))
    data = self.s.recv(1024).decode("utf-8") 
    if json.loads( data)['error']=='1':
         self.s.close()
         raise Exception('Error while getting context variable')  
    self.s.close()    
    return (json.loads(data)['value'])
    
      
    

  